### PR TITLE
Controller tests

### PR DESCRIPTION
**Pull request by ajm188**
Add tests for some edge cases that shouldn't really be covered in feature testing.

### CODE REVIEW

File: spec/controllers/wishlist_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,158 @@
+require 'rails_helper'
+
+RSpec.describe WishlistController, type: :controller do
+  let(:cas_session) { {cas_user: 'bob'} }
```
Comment (ajm188) [inline spec/controllers/wishlist_controller_spec.rb]: how would we ever tell the difference between the hash and the block though

File: spec/controllers/users_controller_spec.rb [hunk 2]
```diff
@@ -1,4 +1,28 @@
...
+  describe 'GET #login' do
+    before { fake_cas }
+
+    it 'redirects to root path by default' do
+      get :login
+      expect(response).to redirect_to root_path
+    end
+
+    it 'redirects to the specified url' do
+      url = '/somewhere'
```
Comment (StephHippo) [inline spec/controllers/users_controller_spec.rb]: Wait, does this work? That's not a legit url, right? Can we just get a comment or something here? If I'm confused and I have some idea of what's going on, any newcomers would definitely be confused.

File: spec/controllers/wishlist_controller_spec.rb (hunk 1 above)
Comment (StephHippo) [inline spec/controllers/wishlist_controller_spec.rb]: #hashrocketsyntax4lyfe

File: spec/controllers/wishlist_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+  let(:wishlist_item) { double('wishlist_item') }
+  let(:relation) { double(first_or_initialize: wishlist_item) }
+
+  describe 'POST #add_course' do
+    context 'when the item can be saved' do
+      before { allow(wishlist_item).to receive(:save).and_return true }
```
Comment (StephHippo) [inline spec/controllers/wishlist_controller_spec.rb]: I honestly don't know what this does either.

Comment (StephHippo) [inline spec/controllers/wishlist_controller_spec.rb]: oooooooooh rspec thing. Wow, you can tell I don't write tests in Ruby daily anymore. Sorry. Ignore these comments. For posterity's sake though: https://www.relishapp.com/rspec/rspec-mocks/v/2-99/docs/method-stubs/allow-with-a-simple-return-value

**Comment by StephHippo** [comment]
:+1:

File: spec/controllers/wishlist_controller_spec.rb (hunk 1 above)
Comment (ajm188) [inline spec/controllers/wishlist_controller_spec.rb]: #getoffmylawn

File: spec/controllers/users_controller_spec.rb (hunk 2 above)
Comment (ajm188) [inline spec/controllers/users_controller_spec.rb]: This actually does work. `redirect_to` constructs a 302 response to wherever you tell it to go, so you can also have it redirect to completely different domains. Should I have it be an actual URL in the project?

File: spec/controllers/wishlist_controller_spec.rb (hunk 3 above)
Comment (ajm188) [inline spec/controllers/wishlist_controller_spec.rb]: Yeah, maybe some more detail for posterity:

`allow(thing).to receive(:method).and_return value ` stubs the method on thing, so that when it is called it returns the specified value instead of calling the actual method on the object. There a whole bunch of other cool things that you can do with stubbing, which you can read in the docs :). It's used here because it's a really great way to make saving objects fail without having to seed them with data that will fail validations (logic which would need to be updated any time the validations on that object change).

**Comment by ajm188** [comment]
How do I miss the +1 comments?
